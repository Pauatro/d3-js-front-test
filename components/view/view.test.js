import createView from './view.js'

it('renders correctly', () => {

    const container = document.createElement('div')
    container.style = "width: 300px; height: 200px;"

    document.body.append(container)

    const data = {
        "name": "revenue",
        "units": "â‚¬",
        "total": 200000,
        "timeLine": [
            100,
            110,
            105,
            120,
            118,
            140,
            135,
            140,
            135
        ],
        "grouped": [
            {
                "group": "tablet",
                "perc": 60,
                "total": 120000
            },
            {
                "group": "smartphone",
                "perc": 40,
                "total": 80000
            }
        ]

    }

    createView(data, 0, container)
    expect(container).toMatchSnapshot();
});