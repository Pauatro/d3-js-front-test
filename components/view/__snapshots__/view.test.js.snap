// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style="width: 300px; height: 200px;"
>
  <div
    class="view-container"
  >
    <svg
      viewBox="0,0,0,0"
    >
      <g
        transform="translate(0,2)"
      >
        <path
          d="M0,0L0,0Z"
          style="
            stroke: #93d55d;
            stroke-width: 1px;
            fill: #93d55d;
            fill-opacity: 0.2;
        "
        />
      </g>
      <g
        transform="translate(0,0)"
      >
        <path
          d="M0,0Z"
          fill="#3a6815"
        />
        <path
          d="M0,0Z"
          fill="#93d55d"
        />
        <path
          d="M0,0Z"
          fill="#b3b3b3"
        />
        <path
          d="M0,0Z"
          fill="white"
        />
        <path
          d="M0,0Z"
          fill="white"
        />
        <path
          d="M0,0Z"
          fill="white"
        />
        <path
          d="M0,0Z"
          fill="white"
        />
        <path
          d="M0,0Z"
          fill="white"
        />
      </g>
      <g>
        <text>
          <tspan
            class="central-title"
            dy="0"
            style="
            font-size: 0px;
            stroke-width: 0.5px;
            fill: #b3b3b3;
            stroke: #b3b3b3;
            text-anchor: middle;
            dominant-baseline: middle;
            font-weight: lighter;
            letter-spacing: 1px;
        "
          >
            REVENUE
          </tspan>
        </text>
      </g>
      <g>
        <text
          height="0"
          width="0"
        >
          <tspan
            dy="0"
            style="
            font-size: 0px;
            text-anchor: middle;
            dominant-baseline: middle;
            stroke-width: 0.7px;
            fill: #302c2c;
            stroke: #302c2c;
        "
          >
            200.000€
          </tspan>
        </text>
      </g>
      <foreignobject
        height="0"
        transform="translate(0, 0)"
        width="0"
      >
        <div
          class="table-container"
          style="
            width: 0px;
            height: 0px;
            display: flex;
            align-items: flex-start;
            justify-content: center;
        "
        >
          <table
            style="
            width: 0px;
        "
          >
            <thead
              class="table-header"
            >
              <tr>
                <th
                  style="
            width: 50%;
            text-align: left;
            color: #3a6815;
            font-size: 0px;
        "
                >
                  Tablet
                </th>
                <th
                  style="
            width: 50%;
            text-align: right;
            color: #93d55d;
            font-size: 0px;
        "
                >
                  Smartphone
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  align="left"
                  class="table-body"
                  style="
            width: 50%;
            padding-top: 0px;
            padding-bottom: 0px;
            margin: 0px;
            border-spacing: 0px;
        "
                >
                  <td
                    align="left"
                    class="table-body-left"
                    style=" 
            font-size: 0px; 
            padding-right: 10px;
        "
                  >
                    60%
                  </td>
                  <td
                    align="left"
                    class="table-body-right"
                    style=" 
            font-size: 0px; 
            color: #b3b3b3;        
        "
                  >
                    120.000€
                  </td>
                </td>
                <td
                  align="right"
                  class="table-body"
                  style="
            width: 50%;
            padding-top: 0px;
            padding-bottom: 0px;
            margin: 0px;
            border-spacing: 0px;
        "
                >
                  <td
                    align="right"
                    class="table-body-left"
                    style=" 
            font-size: 0px; 
            padding-right: 10px;
        "
                  >
                    40%
                  </td>
                  <td
                    align="right"
                    class="table-body-right"
                    style=" 
            font-size: 0px; 
            color: #b3b3b3;        
        "
                  >
                    80.000€
                  </td>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </foreignobject>
    </svg>
  </div>
</div>
`;
